src/mam4xx/mo_sethet.hpp

```diff
@@ -176,6 +183,303 @@ void find_ktop(
+  constexpr Real xph0 = 1.0e-5;    // cloud [h+]
+  constexpr Real satf_hno3 = .016; // saturation factor for hno3 in clouds
+  constexpr Real satf_h2o2 = .016; // saturation factor for h2o2 in clouds
+  constexpr Real satf_so2 = .016;  // saturation factor for so2 in clouds
+  const Real const0 = boltz_cgs * 1.0e-6; // [atmospheres/deg k/cm^3]
+  constexpr Real hno3_diss = 15.4;        // hno3 dissociation constant
+  constexpr Real mass_air = 29.0;         // mass of background atmosphere [amu]
+  constexpr Real km2cm = 1.0e5;           // convert km to cm
+  constexpr Real m2km = 1.0e-3;           // convert m to km
+  constexpr Real m3_2_cm3 = 1.0e6;        // convert m^3 to cm^3
+  constexpr Real MISSING = -999999.0;
+  constexpr Real large_value_lifetime =
+      1.0e29; // a large lifetime value if no washout
+
+  int ktop;  // tropopause level, 100mb for lat < 60 and 300mb for lat > 60
+  Real xkgm; // mass flux on rain drop
+  Real stay; // fraction of layer traversed by falling drop in timestep delt
+  Real xdtm; // the traveling time in each dz [s]
+  Real xxx2, xxx3;   // working variables for h2o2 (2) and so2 (3)
+  Real yso2, yh2o2;  // washout lifetime [s]
+  Real work1, work2; // working variables
+  Real zsurf;        // surface height [km]
+
+  //-----------------------------------------------------------------
+  //        note: the press array is in pascals and must be
+  //              mutiplied by 10 to yield dynes/cm**2.
+  //-----------------------------------------------------------------
+  //       ... set wet deposition for
+  //           1. h2o2         2. hno3
+  //           3. ch2o         4. ch3ooh
+  //           5. pooh         6. ch3coooh
+  //           7. ho2no2       8. onit
+  //           9. mvk         10. macr
+  //          11. c2h5ooh     12. c3h7ooh
+  //          13. rooh        14. ch3cocho
+  //          15. pb          16. macrooh
+  //          17. xooh        18. onitr
+  //          19. isopooh     20. ch3oh
+  //          21. c2h5oh      22. glyald
+  //          23. hyac        24. hydrald
+  //          25. ch3cho      26. isopno3
+  //-----------------------------------------------------------------
+  // FORTRAN refactor note: current MAM4 only have three species in default:
+  // 'H2O2','H2SO4','SO2'.  Options for other species are then removed
+  //-----------------------------------------------------------------
+
+  Kokkos::parallel_for(Kokkos::TeamThreadRange(team, local_pver), [&](int kk) {
+    Kokkos::parallel_for(
+        Kokkos::ThreadVectorRange(team, gas_pcnst), [&](int mm) {
+          het_rates[mm](kk) = 0.0;
+          tmp_hetrates[mm](kk) = 0.0; // initiate temporary array
+        });
+  });
+
+  for (int mm = 0; mm < gas_wetdep_cnt; mm++) {
+    int mm2 = wetdep_map[mm];
+    if (mm2 < 0) {
+      Kokkos::parallel_for(Kokkos::TeamVectorRange(team, local_pver),
+                           [&](int kk) { het_rates[mm2](kk) = MISSING; });
```

Comment:
It seems `mm2` is used as an index to `het_rates` if `mm2` is negative.  Maybe it should be `mm` as the index?

Reply:
Thanks for noticing that! The inequality needs to flip to match the fortran